src/libsyntax/parse/lexer/mod.rs

```diff
@@ -1408,9 +1408,10 @@ impl<'a> StringReader<'a> {
                     // lifetimes shouldn't end with a single quote
```

Comment:
Ah! I think I figured it out. If you look at line 1406 I think we might be in this code branch, which doesn't handle that much.

Can you modify the diagnostic message of the different `err_span_` to verify where each error is coming from?